src/engine/GameObjects/MapNode.cxx

```diff
@@ -287,6 +287,19 @@ const MapNodeData &MapNode::getActiveMapNodeData() const
   return m_mapNodeData[Layer::TERRAIN];
 }
 
+void MapNode::setMapNodeData(std::vector<MapNodeData> mapNodeData)
```

Comment:
why are you passing this by copy?

Reply:
I think you could do one of the following to improve this API
1. Send the vector by rvalue
2. Expose an accessor instead (like `operator[]` or something)
3. Using an iterator template and swapping each pair of elements
These new designs would allow us to get rid of an unnecessary copy

Reply:
unfortunately, we get a deserialization object here from the json library. passing it as ref doesn't work. 

Since this function is still fast enough imho i'd let it as it is.

Reply:
Ah, it works as rvalue ref.